lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,76 @@
+Facter.add(:powershell_version) do
+  setcode do
+    if Puppet::Util::Platform.windows?
+      version = Facter.value(:powershell_three_version) || Facter.value(:powershell_one_version)
```

Comment:
But I would also double check to make sure that Facter doesn't turn it into a `string` by the time we use it in our `:confine`.